xbmc/platform/win32/filesystem/Win32SMBDirectory.cpp

```diff
@@ -158,35 +182,36 @@ bool CWin32SMBDirectory::GetDirectory(const CURL& url, CFileItemList &items)
       continue;
     }
 
-    CFileItemPtr pItem(new CFileItem(itemName));
-
-    pItem->SetFolder((findData.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY) != 0);
-    if (pItem->IsFolder())
-      pItem->SetPath(pathWithSlash + itemName + '/');
-    else
-      pItem->SetPath(pathWithSlash + itemName);
+    const bool isDir = (findData.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY) != 0;
+    const bool isHidden =
+        (findData.dwFileAttributes & (FILE_ATTRIBUTE_HIDDEN | FILE_ATTRIBUTE_SYSTEM)) != 0 ||
+        itemName.front() == '.'; // mark files starting from dot as hidden
 
-    if ((findData.dwFileAttributes & (FILE_ATTRIBUTE_HIDDEN | FILE_ATTRIBUTE_SYSTEM)) != 0
-          || itemName.front() == '.') // mark files starting from dot as hidden
-      pItem->SetProperty("file:hidden", true);
+    std::string itemPath(pathWithSlash + itemName);
+    if (isDir)
+      itemPath += '/';
 
     // calculation of size and date costs a little on win32
     // so DIR_FLAG_NO_FILE_INFO flag is ignored
-    FileTime localTime{};
-    if (FileTimeToLocalFileTime(reinterpret_cast<FileTime*>(&findData.ftLastWriteTime),
-                                &localTime) == TRUE)
-      pItem->SetDateTime(localTime);
+    std::shared_ptr<CFileItem>& item =
+        fileItems.emplace_back(std::make_shared<CFileItem>(itemName));
+    item->SetFolder(isDir);
+    item->SetPath(itemPath);
+    if (auto entryTime = GetEntryTime(findData); entryTime)
+      item->SetDateTime(std::move(*entryTime));
     else
     {
-      CDateTime dt{pItem->GetDateTime()};
+      CDateTime dt{item->GetDateTime()};
       dt.SetValid(false);
-      pItem->SetDateTime(dt);
+      item->SetDateTime(std::move(dt));
     }
```

Comment:
For me this needs to go, it accomplishes nothing and complicates this PR.
The code before the last refactor was:
    if (KODI::TIME::FileTimeToLocalFileTime(&fileTime, &localTime) == TRUE)
      pItem->m_dateTime = localTime;
    else
      pItem->m_dateTime.SetValid(false);
which already did nothing, since pItem is new and its m_dateTime is default, which is in invalid status. And I think that can be relied on.